lib/rmg/CRmgTemplateZone.cpp

```diff
@@ -1335,12 +1355,24 @@ bool CRmgTemplateZone::placeMines ()
 	}
 	for (int i = 0; i < mines[Res::GOLD]; i++)
 	{
-		auto mine = (CGMine *) factory.at(Res::GOLD)->create(ObjectTemplate());
+		auto mine = (CGMine *) factoryMine.at(Res::GOLD)->create(ObjectTemplate());
+        tempmines.emplace_back(mine);
 		mine->producedResource = Res::GOLD;
 		mine->tempOwner = PlayerColor::NEUTRAL;
 		mine->producedQuantity = mine->defaultResProduction();
 		addRequiredObject(mine, 7000);
 	}
+    
+    //create extra resources
+    for(auto* mine : tempmines)
+    {
+        for(int rc = gen->rand.nextInt(1,3); rc>0; --rc)
+        {
+            auto resourse = (CGResource*) factoryRes.at(mine->producedResource)->create(ObjectTemplate());
+            resourse->amount = 0;
```

Comment:
Please add comment what amount = 0 means;

Reply:
Or consider introducing a constant CGResource::SOMETHING_AMOUNT = 0 or a method in CGResource like setSomethingAmount and comment will be no longer needed.

Reply:
fixed